### PR TITLE
Read and store ST configuration in yaml

### PR DESCRIPTION
**Pull request by kornys**
### Type of change

- Enhancement / new feature

### Description

Read and store configuration from environment variables for systemtest run in yaml format.
Configuration is automatically created for every systemtest run so test run can be easily reproduces with just passing env file and not writing all env vars.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Make sure all tests pass
- [x] Update documentation

### CODE REVIEW

**Review by Frawless — APPROVED** [review]

File: development-docs/TESTING.md [hunk 1]
```diff
@@ -295,12 +295,17 @@ All available test groups are listed in [Constants](systemtest/src/main/java/io/
...
 
 Variables can be defined via environmental variables or a configuration file; this file can be located anywhere on the file system as long as a path is provided to this file.
-The path is defined by the environmental variable `ST_CONFIG_PATH`; if the `ST_CONFIG_PATH` environmental variable is not defined, the default config file location is used `systemtest/config.json`.
+The path is defined by the environmental variable `ST_CONFIG_PATH`; if the `ST_CONFIG_PATH` environmental variable is not defined, the default config file location is used `systemtest/config.yaml`.
 Loading of system configuration has the following priority order:
 1. Environment variable
 2. Variable defined in the configuration file
 3. Default value
 
+After every systemtest tes run all env variables is automatically stored into $TEST_LOG_DIR/test-run.../config.yaml so it test run can be easily reproduced just by running
```
Comment (im-konge) [inline development-docs/TESTING.md:304]:
```suggestion
After every systemtest test run all env variables are automatically stored into $TEST_LOG_DIR/test-run.../config.yaml so each test run can be easily reproduced just by running
```

File: systemtest/src/main/java/io/strimzi/systemtest/Environment.java [hunk 2]
```diff
@@ -370,16 +379,37 @@ private static <T> T getOrDefault(String var, Function<String, T> converter, T d
...
+        LinkedHashMap<String, String> toSave = new LinkedHashMap<>();
+
+        VALUES.forEach((key, value) -> {
+            if (isWriteable(key, value)) {
+                toSave.put(key, value);
+            }
+        });
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        mapper.writerWithDefaultPrettyPrinter().writeValue(logPath.resolve("config.yaml").toFile(), toSave);
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/Environment.java:395]: Maybe `config.yaml` can be constant either here or in `TestConstants`?

**Review by im-konge — COMMENTED** [review]
Just two nits, thanks for the PR :)

File: systemtest/src/main/java/io/strimzi/systemtest/Environment.java (hunk 2 above)
Comment (kornys) [reply systemtest/src/main/java/io/strimzi/systemtest/Environment.java:395]: I don't think it will be used somewhere else so I would keep it in string.

**Review by im-konge — APPROVED** [review]

**Review by see-quick — APPROVED** [review]

**Pull request merged**